proxy/src/telemetry/metrics/labels.rs

```diff
@@ -392,3 +576,51 @@ impl Into<ctx::transport::TlsStatus> for TlsStatus {
         self.0
     }
 }
+
+// ===== impl TlsConfigLabels =====
+
+impl TlsConfigLabels {
+    pub fn success() -> Self {
+        TlsConfigLabels::Reloaded
+    }
+}
+
+impl From<tls::ConfigError> for TlsConfigLabels {
+    fn from(err: tls::ConfigError) -> Self {
+        match err {
+            tls::ConfigError::Io(path, error_code) =>
+                TlsConfigLabels::Io { path, errno: error_code.map(Errno::from) },
+            tls::ConfigError::FailedToParseTrustAnchors(_) =>
+                TlsConfigLabels::InvalidTrustAnchors,
+            tls::ConfigError::EndEntityCertIsNotValid(_) =>
+                TlsConfigLabels::InvalidEndEntityCert,
+            tls::ConfigError::InvalidPrivateKey =>
+                TlsConfigLabels::InvalidPrivateKey,
+        }
+    }
+}
+
+impl fmt::Display for TlsConfigLabels {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            TlsConfigLabels::Reloaded =>
+                f.pad("status=\"reloaded\""),
+            TlsConfigLabels::Io { ref path, errno: Some(errno) } =>
+                write!(f,
+                    "status=\"io_error\",path=\"{}\",errno=\"{}\"",
+                    path.display(), errno
+                ),
+            TlsConfigLabels::Io { ref path, errno: None } =>
+                write!(f,
+                    "status=\"io_error\",path=\"{}\",errno=\"UNKNOWN\"",
+                    path.display(),
+                ),
+            TlsConfigLabels::InvalidPrivateKey =>
+                f.pad("status=\"invalid_private_key\""),
+            TlsConfigLabels::InvalidEndEntityCert =>
+                f.pad("status=\"invalid_end_entity_cert\""),
+            TlsConfigLabels::InvalidTrustAnchors =>
+                f.pad("status=\"invalid_trust_anchors\""),
```

Comment:
OK. All the escaped double quotes make my eyes glaze over. Maybe we should switch to raw string literals in a follow-up PR so that these kinds of changes are easier to read/verify.

Reply:
I think we can use raw string literals in places where we call `Formatter::pad` with a string constant. However, we won't be able to use them as format strings, so we'll still need to escape double quotes in places where we use `write!`. I'm not sure if mixing the two will make the code more or less readable, WDYT?

Reply:
> However, we won't be able to use them as format strings

I did not know that.

Reply:
Nevermind, I was wrong about that, it seems we can use raw strings as format strings. Just tried it in the playground and it worked. I recall this not being possible at one point, but this doesn't seem to be the case any longer.